### PR TITLE
mono explosions

### PR DESCRIPTION
**Pull request by bend-n**
explodes monos for a constant resource increment.

### CODE REVIEW

File: src/main/PlagueMain.java [hunk 1]
```diff
@@ -522,6 +570,10 @@ public void registerClientCommands(CommandHandler handler) {
             endgame(new Seq<>(), args);
         });
 
+        handler.<Player>register("monos", "explain monos", (args, player) -> {
+            Call.infoMessage(player.con, mono_info);
+        });
+
```
Comment (nilederg) [inline src/main/PlagueMain.java:584]: bundle into help

File: src/main/PlagueMain.java [hunk 2]
```diff
@@ -475,6 +491,31 @@ public void init() {
...
+                }
+            }
+        });
+        // monos must die
+        Events.on(EventType.UnitCreateEvent.class, event -> {
+            if (event.unit.type == UnitTypes.mono) {
+                event.unit.health = 0;
+                event.unit.dead = true;
+            }
+        });
```
Comment (nilederg) [inline src/main/PlagueMain.java:525]: alert player to when monos are capped out

**Review by nilederg — CHANGES_REQUESTED** [review]
As detailed in comments, please bundle mono explanation into help (575), and alert the player as to when they have produced all possible monos (500). Additionally, as per OOP principles, I request you declare the mono cap as a `final byte` in `PlagueTeam.java`

**Comment by bend-n** [comment]
doesnt final mean const?

**Comment by nilederg** [comment]
yes final is java for const

**Comment by bend-n** [comment]
but it isnt const

**Comment by nilederg** [comment]
oh no you're right, it should be const. apologies

File: src/main/PlagueMain.java (hunk 1 above)
Comment (bend-n) [reply src/main/PlagueMain.java:584]: why though? when people say, why mono explode, if you tell them to run `/info`, they're not going to read the entire thing to the end?
its easier to tell them to run `/monos`.

File: src/main/PlagueMain.java [hunk 3]
```diff
@@ -93,15 +94,16 @@ public class PlagueMain extends Plugin {
...
             " the [scarlet]Plague [accent]and [green]Survivors[accent].\n\n" +
             "The [scarlet]Plague[accent] build up their economy to make the biggest army possible, and try to" +
             " break through the [green]Survivors[accent] defenses.\n\n" +
             "The [green]Survivors[accent] build up a huge defense and last 45 minutes to win.\n\n" +
             "To become a " +
             "[green]Survivor[accent], you must place a core in the first 2 minutes of the game, where you are " +
             "allowed to choose your team. Place any block to place a core at the start of the game.\n\n" +
-            "Air units do no damage before 45 minutes";
+            "Air units do no damage before 45 minutes\n\n";
+    final String mono_info = "Monos() will explode, adding to your teams internal mono pool. Abstract monos will then produce 120 copper & lead per second, with a limit of 64 \"monos\"";
```
Comment (nilederg) [inline src/main/PlagueMain.java]: const

File: src/main/PlagueMain.java [hunk 4]
```diff
@@ -93,15 +94,16 @@ public class PlagueMain extends Plugin {
 
     public int counts;
 
-    String info = "[#4d004d]{[purple]AA[#4d004d]} [olive]Plague[accent] is a survival game mode with two teams," +
+    final String info = "[#4d004d]{[purple]AA[#4d004d]} [olive]Plague[accent] is a survival game mode with two teams," +
```
Comment (nilederg) [inline src/main/PlagueMain.java:98]: const

File: src/main/PlagueMain.java [hunk 5]
```diff
@@ -299,6 +305,16 @@ public void init() {
...
+            // runs 20 times a second, 1 mono = 20/s. caps at 255 monos = 5100/s
+            if (Core.graphics.getFrameId() % 4 == 0) {
+                for (PlagueTeam team : teams.values()) {
+                    if (team.monos > 0 && !team.team.cores().isEmpty()) {
+                        CoreBuild core = team.team.cores().first();
+                        core.items.add(Items.copper, team.monos);
+                        core.items.add(Items.lead, team.monos);
+                    }
+                }
+            }
```
Comment (nilederg) [inline src/main/PlagueMain.java:317]: hilarious output thank you

File: src/main/PlagueMain.java (hunk 1 above)
Comment (nilederg) [reply src/main/PlagueMain.java:584]: fair enough.

File: src/main/PlagueMain.java (hunk 4 above)
Comment (bend-n) [reply src/main/PlagueMain.java:98]: ??? const is a keyword not?

File: src/main/PlagueMain.java (hunk 2 above)
Comment (nilederg) [reply src/main/PlagueMain.java:525]: pls capitalize and add apostrophe

**Review by nilederg — CHANGES_REQUESTED** [review]
refer to requests

File: src/main/PlagueMain.java (hunk 1 above)
Comment (bend-n) [reply src/main/PlagueMain.java:584]: I bundled it into /info while keeping /monos.

File: src/main/PlagueMain.java (hunk 4 above)
Comment (nilederg) [reply src/main/PlagueMain.java:98]: fine no static

**Review by nilederg — APPROVED** [review]
Nice feature. 5100/s may be broken but it's certainly funny. Thank you, and LGTM.

**Comment by nilederg** [comment]
Merge at will.

**Pull request merged**